back/src/chat/channel/channel.service.ts

```diff
@@ -12,143 +8,152 @@ import {
-          console.log('5: User is banned');
-          throw new ForbiddenException('Cannot join channel', {
-            cause: new Error(),
-            description: 'User is banned',
-          });
-        }
-        if (channelConnection.role !== ChannelRole.INVITED) {
-          throw new BadRequestException('Cannot join channel', {
-            cause: new Error(),
-            description: 'User already in channel',
-          });
-        }
-        console.log('4: User is invited');
-        await this.joinChannel(user, channel);
-      } else {
-        if (this.isChannelProtected(channel)) {
-          console.log('6: Channel is protected');
-          //TODO: Make this a real function please
-          if (this.isPasswordOk(channel, data)) {
-            console.log('7: Password is ok');
-            await this.joinChannel(user, channel);
-          } else {
-            console.log('7: Wrong password.');
-            throw new ForbiddenException('Cannot join channel', {
-              cause: new Error(),
-              description: 'Incorrect password',
-            });
-          }
-        }
-        if (this.isChannelPrivate(channel)) {
-          console.log('8: Channel is private');
-          throw new ForbiddenException('Cannot join channel', {
-            cause: new Error(),
-            description: 'User not on invite list',
-          });
-        } else {
-          console.log('9: Default joining channel');
-          await this.joinChannel(user, channel);
-        }
-      }
-    } else {
-      console.log('10: Channel does not exist, creating it');
-      await this.createChannel(user, data);
-    }
+  async getAllChannelConnections(
+    pagination: PaginationDto,
+  ): Promise<ChannelConnection[]> {
+    return await this.prisma.channelConnection.findMany(pagination);
   }
 
-  isChannelProtected(channel: Channel): boolean {
-    if (channel.kind === ChannelKind.PROTECTED) return true;
-    return false;
+  async getChannelConnection(
+    userId: number,
+    channelId: number,
+  ): Promise<ChannelConnection> {
+    return (
+      await this.prisma.channelConnection.findMany({
+        where: {
+          AND: [{ userId }, { channelId }],
+        },
+      })
+    )[0];
   }
 
-  isChannelPrivate(channel: Channel): boolean {
-    if (channel.kind === ChannelKind.PRIVATE) return true;
-    return false;
+  async getChannelConnections(
+    channelId: number,
+    pagination: PaginationDto,
+  ): Promise<ChannelConnection[]> {
+    return await this.prisma.channelConnection.findMany({
+      where: { channelId },
+      ...pagination,
+    });
   }
 
-  //TODO: Fix this garbage function
-  isPasswordOk(channel: Channel, data: ChannelDto): boolean {
-    if (channel.password === data.password) return true;
-    return false;
+  async isUserInChannel(userId: number, channelId: number): Promise<boolean> {
+    return !!(await this.getChannelConnection(userId, channelId));
```

Comment:
Something like that maybe?
```suggestion
    const channelConnection = await this.getChannelConection(userId, channelId));
    if (!channelConnection || channelConnection.role === channelRole.INVITED
      || channelConnection.role ==== channelRole.BANNED)
      return false
    return true;
```

Reply:
updated in : [4da690d](https://github.com/42rave/transcendence/pull/36/commits/4da690d473b70ada9865cb0e1fdf933ac74956ca)